src/cdomains/intDomain.ml

```diff
@@ -524,7 +524,30 @@ struct
 
   let norm ik = function None -> None | Some (x,y) ->
     if Ints_t.compare x y > 0 then None
-    else if Ints_t.compare (min_int ik) x > 0 || Ints_t.compare (max_int ik) y < 0 then (set_overflow_flag ik; top_of ik)
+    else if Ints_t.compare (min_int ik) x > 0 || Ints_t.compare (max_int ik) y < 0 then (
+      set_overflow_flag ik;
+      if should_wrap ik then
+        (* We can only soundly wrap if at most one overflow occurred, otherwise the minimal and maximal values of the AP interval *)
```

Comment:
`AP interval`?

Reply:
Arbitrary Precision, I borrowed some LLVM terminology here without intending to.